File: src/Simulator.java
/**
 * The main class for simulation.
 *
 * @author Dafu Ai, Wenzhou Wei
 */
public class Simulator {

    /**
     * Runs the simulation
     */
    public static void main(String[] args) throws InterruptedException {
        int tick = 0;
        World world = new World();

        while (true) {
            System.out.println("Frame #" + tick);
            world.update();
            System.out.println("");

            Thread.sleep(1000);
            tick ++;
        }
    }
}

File: rebellion-python/simulator.py
from time import sleep
from models import World

# Author: Dafu Ai
from static_params import MAX_FRAMES


def main():
    """The entry point for simulation."""
    frame = 1
    world = World(dynamic_params_filename='dynamic_params.json', output_filename='out.csv')

    while frame <= MAX_FRAMES:
        print("Frame #" + str(frame))
        world.update(frame)

        sleep(1)
        frame += 1


if __name__ == '__main__':
    main()
